gk/main.c

```diff
@@ -865,21 +1019,41 @@ process_pkts_front(uint8_t port_front, uint8_t port_back,
 			switch (fib->action) {
 			case GK_FWD_GRANTOR:
 				/*
+				 * We heuristically drop entries to
+				 * alleviate memory pressure
+				 * when the table is full.
+				 *
 				 * The entry instructs to enforce
 				 * policies over its packets,
 			 	 * initialize an entry in the
 				 * flow table, proceed as the
 				 * brand-new entry instructs, and
 			 	 * go to the next packet.
 			 	 */
-				ret = rte_hash_add_key_with_hash(
-					instance->ip_flow_hash_table,
- 					&packet.flow, pkt->hash.rss);
-				if (ret < 0) {
-					RTE_LOG(ERR, HASH,
-						"The GK block failed to add new key to hash table!\n");
+				ret = gk_hash_add_flow_entry(
+					instance, &packet.flow,
+					gk_conf->request_timeout_cycles,
+					pkt->hash.rss, GK_REQUEST);
+				if (ret == -EINVAL) {
```

Comment:
Test for `ret < 0` after testing for `ret == -ENOSPC` instead of testing for `ret == -EINVAL`.